llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp

```diff
@@ -4142,9 +4142,40 @@ LegalizerHelper::LegalizeResult LegalizerHelper::lowerStore(GStore &StoreMI) {
   }
 
   if (MemTy.isVector()) {
-    // TODO: Handle vector trunc stores
-    if (MemTy != SrcTy)
+    LLT MemScalarTy = MemTy.getElementType();
+    if (MemTy != SrcTy) {
+      if (!MemScalarTy.isByteSized()) {
+        // We need to build an integer scalar of the vector bit pattern.
+        // It's not legal for us to add padding when storing a vector.
+        unsigned NumBits = MemTy.getSizeInBits();
+        LLT IntTy = LLT::scalar(NumBits);
+        auto CurrVal = MIRBuilder.buildConstant(IntTy, 0);
+        LLT IdxTy = getLLTForMVT(TLI.getVectorIdxTy(MF.getDataLayout()));
```

Comment:
Probably should just have an LLT version of getVectorIdxTy (I thought we had one already?)

Reply:
I'd rather make that change separately since it'll make sense otherwise convert all of our existing uses of the MVT version over.